File: javascript/day04/part2.js
const data = require('../file-reader.js')
    .readFile('input.txt', '\n', String)
    .map(s => s.split(',').map(s => s.split('-').map(Number)).map(a => ({ s: a[0], e: a[1]})))

function isOverlapping(pair) {
    const [a, b] = pair
    return !(a.s > b.e || b.s > a.e)
}

const sum = data.reduce((sum, pair) => sum + isOverlapping(pair), 0)
console.log(sum)

File: python/day04/part2.py
t = 0

for line in open('input.txt'):
    a, b, x, y = map(int, line.replace(",", "-").split("-"))
    if set(range(a, b + 1)) & set(range(x, y + 1)):
        t += 1

print(t)
assert t == 801, 'Incorrect result'
